compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2441,10 +2441,26 @@ impl<'a> Parser<'a> {
                     self.error_on_extra_if(&cond)?;
                     // Parse block, which will always fail, but we can add a nice note to the error
                     self.parse_block().map_err(|mut err| {
-                        err.span_note(
-                            cond_span,
-                            "the `if` expression is missing a block after this condition",
-                        );
+                        if self.prev_token == token::Semi
+                            && self.token == token::AndAnd
+                            && let maybe_let = self.look_ahead(1, |t| t.clone())
+                            && maybe_let.is_keyword(kw::Let)
```

Comment:
```suggestion
                            && self.look_ahead(1, |t| t.is_keyword(kw::Let))
```

Reply:
I broke the check into two lines mainly to avoid doing something like this later when printing the `let` span (i.e., we can just reuse the `maybe_let`'s span):
```rust
err.span_note(
    self.token.span.to(self.look_ahead(1, |t| t.clone()),
    "you likely meant to continue parsing the let-chain starting here",
 );
```
Though I'm not sure which of the two would actually read better.

Reply:
Ah, I missed you were using the span in the label. That makes sense.